docs/ref/models/relations.txt

```diff
@@ -70,6 +70,9 @@ Related objects reference
         :ref:`intermediate model <intermediary-manytomany>` instance(s), if
         needed.
 
+        ``add()`` also accepts primary keys as arguments, so the above example
+        can be written directly as ``b.entry_set.add(234)``.
+
```

Comment:
I _think_ this is not always the primary key that gets accepted but the field the relation points to.

For example, if `to_field` is defined for the relationship then it's this field value that must be provided.

```python
class Book(models.Model):
    isbn = models.CharField(unique=True)

class LibraryBook(models.Model):
    book = models.ForeignKey(Book, to_field='isbn')
    library  = models.ForeignKey('Library')
    amount = models.SmallPositiveIntegerField()

class Library(models.Model):
    books = ManyToMany(Book, through=LibraryBook)
```

In the above case `library.books` would expect `Book.isbn` values and not `Book.id` values.

The same can be said for remove.

Reply:
Do you know if there are tests for this behaviour? I can't seem to find any (with a simple `grep -r ".add(" tests/m2m_*`, though, I might be missing something?)

Would it be of value to add tests for this behaviour, assuming there are none? And what would be the best place to add them? `m2m_multiple`?

Reply:
I assume some tests exists the current way tests are structured is kind of a mess so it's hard to tell. It might be used indirectly by `forms.ModelMultipleChoiceField`. I suggest you put a break point in https://github.com/django/django/blob/1ffddfc233e2d5139cc6ec31a4ec6ef70b10f87f/django/db/models/fields/related_descriptors.py#L959-L973 based on what `*objs` contains.

Reply:
I wasn't in a pdb mood, so I wrote a test instead (proving you're right, btw). Do you think adding the tests would be a good idea?

Reply:
An additional test can't hurt if none is covering this case already but it'd be great to assert if it's the case of not.

Reply:
Yeah, there are plenty of calls to this method, e.g. from `many_to_many`, `serializers.test_natural.forward_ref_m2m_test`, `serializers.tests.test_forward_refs`, etc, as well as plenty of `loaddata` interaction (probably mostly via test set up).

But since as far as I could see there were no explicit tests for this behaviour, I added some in this PR, including one for mixed additions `obj.something_set.add(obj, 1)`, which I'm pretty sure was covered by no test so far.